### PR TITLE
Create a logs@custom index template reference

### LINKED ISSUE #3393: [Logs+] Create a logs@custom reference
In reference to [PR 102456](https://github.com/elastic/elasticsearch/pull/102456), we need to document how to set the `default_field` to specific fields using the `logs@custom` component template.

To do this, we can create a reference page explaining how to use the template. 

To start, look into current documentation on that mentions the [*@custom](https://github.com/search?q=org%3Aelastic+%40custom+language%3AAsciiDoc&type=code&l=AsciiDoc) component templates and check with Ingest for potential crossover.

Overview document [HERE](https://docs.google.com/document/d/1DDcA54YYpJx23pGC_tqy4fO9Ome3zubYTMdU9d1zyLU/edit#heading=h.j2hpon7h872n)

### PR DESCRIPTION
**Pull request by mdbirnstiehl**
this PR closes #3393 

Create a log@custom index template guide that also references how to update the `default_field` to `message` after an update has changed the default behavior to `[*]`.

### CODE REVIEW

**Comment by mdbirnstiehl** [comment]
@eyalkoren @ruflin I've created this PR as a first iteration for the logs@custom reference to show how to update the `default_field`. Potentially, we can include other common uses. I think it would be helpful to add the "why" users might want to use the `message` field over the `[*]`, but I was unsure of that.

File: docs/en/observability/logs-index-template.asciidoc [hunk 1]
```diff
@@ -0,0 +1,54 @@
+[[logs-index-template]]
+= Logs index template reference
+
+[[custom-logs-index-template]]
+== View index templates
+
+Index templates are used to configure the backing indices of data streams as they are created.
+These index templates are composed of multiple component templates--reusable building blocks
```
Comment (ruflin) [inline docs/en/observability/logs-index-template.asciidoc:8]: Can we link to the Elasticsearch docs for component templates here for users that are interested in more details?

File: docs/en/observability/logs-index-template.asciidoc [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+* `logs@custom`
+* `ecs@mappings`
+
+You can use the `@custom` component template to customize your {es} indices. The `logs@custom` template is not installed by default, but you can create a template named `logs@custom` to override and extend default mappings or settings.
+
+Open {kib} and navigate to **{stack-manage-app}** → **Index Management** → **Component Templates**.
+
+. Click *Create component template*.
+. Name the component template logs@custom.
+. Add any custom metadata, index settings, or mappings.
```
Comment (ruflin) [inline docs/en/observability/logs-index-template.asciidoc:32]: After adding the template, all indices that have it, a rollover needs to be triggered to make the changes available.

**Review by ruflin — COMMENTED** [review]
@zmoog @gsantoro Can we get your review on this?

File: docs/en/observability/logs-index-template.asciidoc [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+The default Logs index template can be viewed in {kib}.
+Navigate to **{stack-manage-app}** → **Index Management** → **Index Templates**, and search for `logs`.
+Select any of the logs index templates to view their relevant component templates.
+
+[discrete]
+[custom-logs-template-edit]
+=== Edit the {es} index template
+
+{es} has a built-in index template for the `logs-*-*` index pattern, which is composed of component templates including:
```
Comment (gsantoro) [inline docs/en/observability/logs-index-template.asciidoc:19]: I would mention that this default index template is called just `logs`

File: docs/en/observability/logs-index-template.asciidoc [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+
+* `logs@mappings`
+* `logs@settings`
+* `logs@custom`
+* `ecs@mappings`
+
+You can use the `@custom` component template to customize your {es} indices. The `logs@custom` template is not installed by default, but you can create a template named `logs@custom` to override and extend default mappings or settings.
```
Comment (gsantoro) [inline docs/en/observability/logs-index-template.asciidoc:26]:
```suggestion
You can use the `logs@custom` component template to customize your {es} indices. The `logs@custom` component template is not installed by default, but you can create a component template named `logs@custom` to override and extend default mappings or settings.
```

File: docs/en/observability/logs-index-template.asciidoc [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+
+. Click *Create component template*.
+. Name the component template logs@custom.
+. Add any custom metadata, index settings, or mappings.
+
+[discrete]
+[[custom-logs-template-default-field]]
+=== Set the `default_field` using the custom template
+
+The `logs-*-*` index template uses `default_field: [*]` meaning queries without specified fields will search across all fields.
```
Comment (gsantoro) [inline docs/en/observability/logs-index-template.asciidoc:38]:
```suggestion
The `logs` index template uses `default_field: [*]` meaning queries without specified fields will search across all fields.
```

`logs-*-*` is the index pattern referenced by the index template `logs`

File: docs/en/observability/logs-index-template.asciidoc (hunk 4 above)
Comment (gsantoro) [inline docs/en/observability/logs-index-template.asciidoc:26]: I currently get this error because logs@custom doesn't exist. I'm using this version of the stack `8.13.0-f56d9bd5-SNAPSHOT` released on 2024-02-05

```json
{
  "statusCode": 400,
  "error": "Bad Request",
  "message": "index_template [simulate_template_cjga1q24s0swusgjolwsba] invalid, cause [index template [simulate_template_cjga1q24s0swusgjolwsba] specifies component templates [logs@custom] that do not exist]",
  "attributes": {
    "error": {
      "root_cause": [
        {
          "type": "invalid_index_template_exception",
          "reason": "index_template [simulate_template_cjga1q24s0swusgjolwsba] invalid, cause [index template [simulate_template_cjga1q24s0swusgjolwsba] specifies component templates [logs@custom] that do not exist]"
        }
      ],
      "type": "invalid_index_template_exception",
      "reason": "index_template [simulate_template_cjga1q24s0swusgjolwsba] invalid, cause [index template [simulate_template_cjga1q24s0swusgjolwsba] specifies component templates [logs@custom] that do not exist]"
    }
  }
}
```

File: docs/en/observability/logs-index-template.asciidoc [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+You can update the `default_field` to  search in the `message` field instead of all fields using the `logs@custom` component template.
+
+Either create the `logs@custom` template as detailed in the previous section, or edit the existing one to include the following in the *Index settings*:
```
Comment (gsantoro) [inline docs/en/observability/logs-index-template.asciidoc:41]:
```suggestion
Either create the `logs@custom` component template as detailed in the previous section, or add the following code in the *Index settings* of the index template `logs`:
```

I don't think we should suggest to edit the index template `logs` to add this default_field. I would scrap anything after `as detailed in the previous section`

**Review by gsantoro — CHANGES_REQUESTED** [review]

File: docs/en/observability/logs-index-template.asciidoc (hunk 4 above)
Comment (gbamparop) [reply docs/en/observability/logs-index-template.asciidoc:26]: I've encountered this at some point (see [thread](https://elastic.slack.com/archives/CNMNXV4RG/p1707391367379259
)) but couldn't replicate it today after testing several versions (didn't test snapshots)

Comment (gsantoro) [reply docs/en/observability/logs-index-template.asciidoc]: @gbamparop I could use someone else trying the latest snapshot to see if it is just me or we need to open a bug about this.

File: docs/en/observability/logs-index-template.asciidoc [hunk 7]
```diff
@@ -1,44 +1,47 @@
...
 [discrete]
 [[custom-logs-template-default-field]]
 === Set the `default_field` using the custom template
 
-The `logs-*-*` index template uses `default_field: [*]` meaning queries without specified fields will search across all fields.
+The `logs` index template uses `default_field: [*]` meaning queries without specified fields will search across all fields.
 You can update the `default_field` to  search in the `message` field instead of all fields using the `logs@custom` component template.
 
-Either create the `logs@custom` template as detailed in the previous section, or edit the existing one to include the following in the *Index settings*:
+Either create the `logs@custom` component template as outlined in the previous section, or add the following code to the *Index settings* of the `logs` index template:
```
Comment (gsantoro) [inline docs/en/observability/logs-index-template.asciidoc:44]: @ruflin, what do you think about this sentence?

Personally, I wouldn't suggest fiddling with the `logs` index template but only changing the custom component template. 

Thoughts? @gbamparop

**Comment by gsantoro** [comment]
@mdbirnstiehl thanks for the changes. 

I just have two questions left on this PR, as soon as we have an answer to them we can merge this PR

File: docs/en/observability/logs-index-template.asciidoc (hunk 7 above)
Comment (gsantoro) [reply docs/en/observability/logs-index-template.asciidoc:44]: ok. sorry it wasn't 100% clear. 

I would rephrase as "If you don't already have a logs@custom component template you can create it manually. Once you have it, you can add the following to the settings section"

Comment (mdbirnstiehl) [reply docs/en/observability/logs-index-template.asciidoc]: Thanks for pointing it out. I updated it to be clear and add steps for updating the template. I guess because of the way the search works, you can't search for anything using the "@" symbol? or it throws the error:
image

Not sure if there's an underlying reason for this, but it does make it pretty difficult to find any custom templates.

Comment (ruflin) [reply docs/en/observability/logs-index-template.asciidoc]: That sounds like a bug especially as we use it in our names. Has anyone reported this? If not, lets do it.

File: docs/en/observability/logs-index-template.asciidoc (hunk 4 above)
Comment (gsantoro) [reply docs/en/observability/logs-index-template.asciidoc]: @gbamparop I have added some more info to this [thread](https://elastic.slack.com/archives/CNMNXV4RG/p1708015894378669?thread_ts=1707391367.379259&cid=CNMNXV4RG) so that someone else can replicate it.

**Review by gsantoro — APPROVED** [review]

**Comment by gsantoro** [comment]
@mdbirnstiehl just to summarize:
- we need to open a bug for Kibana about not being able to search `logs@custom` because of the character `@`. Can you do this?
- about the other Kibana bug about `logs@custom` missing, I have provided more info in the linked thread by @gbamparop . 

Those bugs are unrelated to this PR. so I have approved this PR now.

**Comment by mdbirnstiehl** [comment]
run docs-build

**Review by bmorelli25 — APPROVED** [review]
